common/shared.c

```diff
@@ -1208,6 +1215,8 @@ int ft_finalize(void)
 		msg.msg_iov = &iov;
 		msg.iov_count = 1;
 		msg.addr = remote_fi_addr;
+		if (hints->mode & FI_CONTEXT)
+			msg.context = &ctx;
```

Comment:
Rather than taking branches in the above 2 checks, we can just set the context field in all cases.